tsdb/engine/tsm1/engine.go

```diff
@@ -917,7 +917,9 @@ func (e *Engine) Backup(w io.Writer, basePath string, since time.Time) error {
 	var path string
 	for i := 0; i < 3; i++ {
 		path, err = e.CreateSnapshot()
-		if err != nil {
+		if err == nil {
+			break
+		} else if err != nil {
```

Comment:
Thanks for the improvement!

The `else if err != nil` is unnecessary with your change and possibly clearer as:

```go
switch {
case err == nil:
	break LOOP
case errors.Is(err, ErrSnapshotInProgress):
	backoff := time.Duration(math.Pow(32, float64(i))) * time.Millisecond
	time.Sleep(backoff)
default:
	return err
}
```

Reply:
Good idea. I've changed it